src/frontend/src/binder/expr/function.rs

```diff
@@ -1369,27 +1369,26 @@ impl Binder {
 
         match HANDLES.get(function_name) {
             Some(handle) => handle(self, inputs),
-            None => Err({
+            None => {
                 let allowed_distance = if function_name.len() > 3 { 2 } else { 1 };
 
                 let candidates = FUNCTIONS_BKTREE
                     .find(function_name, allowed_distance)
-                    .map(|(_idx, c)| c);
+                    .map(|(_idx, c)| c)
+                    .join(" or ");
 
-                let mut candidates = candidates.peekable();
-
-                let err_msg = if candidates.peek().is_none() {
-                    format!("unsupported function: \"{}\"", function_name)
-                } else {
-                    format!(
-                        "unsupported function \"{}\", do you mean \"{}\"?",
-                        function_name,
-                        candidates.join(" or ")
-                    )
-                };
+                let function_sig = format!(
+                    "{}({})",
+                    function_name,
+                    inputs.iter().map(|e| e.return_type()).join(", ")
+                );
 
-                not_implemented!(issue = 112, "{}", err_msg).into()
-            }),
+                Err(NoFunction {
+                    sig: function_sig,
+                    candidates: (!candidates.is_empty()).then(|| candidates),
+                }
+                .into())
```

Comment:
Would you try...

```suggestion
                no_function!(
                    candidates = ..
                    "{}({})", func_name, inputs...
                )
```

Reply:
Thanks for the instruction. I didn't figure out how to use the macro. 😄